### PR TITLE
V1 WB-5141 Pytorch Profiler Integration

### PR DESCRIPTION
**Pull request by vwrj**
[WB-5141] PyTorch Profiler Integration

High-level summary: allow users to profile their training/inference code using the PyTorch Profiler API, with a W&B library method that tracks and syncs profile outputs (in trace files) to the backend. 

https://wandb.atlassian.net/browse/WB-5141

Description
-----------

This PR adds `wandb.profiler` to the wandb client API. It exposes the method `wandb.profiler.torch_trace_handler()` that internally tracks any generated trace files and syncs it to the backend. 

Testing
-------
A test that simulates a typical use-case for the Profiler. Random noise (in the shape of a batch of images) is generated, and a toy convolutional model is trained on this noise using the PyTorch Profiler API. The test ensures that the trace files are synced correctly to the backend, and that an exception is raised if the user does not call `wandb.init()` before `wandb.profiler.torch_trace_handler()`.  

Release Notes
-------------

------------- BEGIN RELEASE NOTES ------------------
Users will now be able to call `wandb.profiler.torch_trace_handler()` with the PyTorch Profiler API in order to sync trace files to the backend (separate backend PR is responsible to install the plugin to allow charts to be rendered in the W&B UI.

------------- END RELEASE NOTES --------------------

[WB-5141]: https://wandb.atlassian.net/browse/WB-5141

### CODE REVIEW

File: wandb/sdk/internal/profiler.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+"""
+import os
+
+import wandb
+
+PYTORCH_PROFILER_MODULE = "torch.profiler"
+
+
+def trace():
+    torch_profiler = wandb.util.get_module(PYTORCH_PROFILER_MODULE)
```
[the remark this answers is not in the document]
Comment (vwrj) [reply wandb/sdk/internal/profiler.py]: ah, thanks!

File: tests/test_profiler.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    with pytest.raises(Exception) as e_info:
+        with torch.profiler.profile(
+            schedule=torch.profiler.schedule(wait=1, warmup=1, active=3, repeat=1),
+            on_trace_ready=wandb.profiler.trace(),
+            record_shapes=True,
+            with_stack=True,
+        ) as prof:
+            for step, batch_data in [(0, 0)]:
+                if step >= (1 + 1 + 3) * 1:
+                    break
```
[the remark this answers is not in the document]
Comment (vwrj) [reply tests/test_profiler.py]: yep, deleting.

File: tests/test_profiler.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import wandb
+
+
+@pytest.mark.skipif(
+    sys.version_info >= (3, 9), reason="PyTorch profiler table in 3.9? to verify"
+)
+def test_profiler_without_init():
+    import torch
+
+    with pytest.raises(Exception) as e_info:
```
[the remark this answers is not in the document]
Comment (vwrj) [reply tests/test_profiler.py]: Done -- I've changed it to UsageError.

**Review by raubitsj — APPROVED** [review]
LGTM

**Pull request merged**